core-java-modules/core-java-reflection/src/test/java/com/baeldung/reflection/PersonAndEmployeeReflectionUnitTest.java

```diff
@@ -9,128 +9,171 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 
 public class PersonAndEmployeeReflectionUnitTest {
 
-    // Fields names
     private static final String LAST_NAME_FIELD = "lastName";
     private static final String FIRST_NAME_FIELD = "firstName";
     private static final String EMPLOYEE_ID_FIELD = "employeeId";
+    private static final String EMPLOYEE_TYPE_FIELD = "LABEL";
     private static final String MONTH_EMPLOYEE_REWARD_FIELD = "reward";
 
     @Test
     public void givenPersonClass_whenGetDeclaredFields_thenTwoFields() {
-        // When
-        Field[] allFields = Person.class.getDeclaredFields();
-
-        // Then
-        assertEquals(2, allFields.length);
-
-        assertTrue(Arrays.stream(allFields).anyMatch(field ->
-          field.getName().equals(LAST_NAME_FIELD)
-            && field.getType().equals(String.class))
-        );
-        assertTrue(Arrays.stream(allFields).anyMatch(field ->
-          field.getName().equals(FIRST_NAME_FIELD)
-            && field.getType().equals(String.class))
-        );
+        List<Field> allFields = Arrays.asList(Person.class.getDeclaredFields());
+
+        assertEquals(2, allFields.size());
+        Field lastName = allFields.stream()
+            .filter(field -> field.getName()
+                .equals(LAST_NAME_FIELD))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Field not found"));
+        assertEquals(lastName.getType(), String.class);
```

Comment:
String.class is the expected value in these assertions and should be the first argument to assertEquals. This occurs several time throughout the tests so I won't point them all out individually.